Mathlib/Combinatorics/SimpleGraph/Connectivity/WalkCounting.lean

```diff
@@ -169,6 +171,35 @@ instance : Decidable G.Connected := by
 instance instDecidableMemSupp (c : G.ConnectedComponent) (v : V) : Decidable (v ∈ c.supp) :=
   c.recOn (fun w ↦ decidable_of_iff (G.Reachable v w) $ by simp)
     (fun _ _ _ _ ↦ Subsingleton.elim _ _)
+
+lemma odd_card_odd_components (ho : Odd (Fintype.card V)) :
+    Odd (Nat.card ({(c : ConnectedComponent G) | Odd (Nat.card c.supp)})) := by
+  simp_rw [Nat.odd_iff_not_even]
+  by_contra! hc
+  simp_rw [← Nat.odd_iff_not_even] at hc
+  rw [
+    ← (set_fintype_card_eq_univ_iff _).mpr G.iUnion_connectedComponentSupp,
+    ← Set.toFinset_card, Nat.odd_iff_not_even] at ho
+  apply ho
+  have : Set.toFinset (⋃ (x : ConnectedComponent G), ConnectedComponent.supp x) =
```

Comment:
separate out as a lemma `toFinset_iUnion`

Reply:
I've tried to, but it seems this statements really needs the `instDecidableMemSupp` above to work out.

This is what I've got, but I don't know how to add a "variable instance" like `[Decidable v ∈f x]` for all v, x.
```
lemma toFinset_iUnion {W} [Fintype W] [Fintype V] (f : W → Set V) : Set.toFinset (⋃ (x : W), f x) =
    Finset.biUnion (Finset.univ : Finset W) (fun x => (f x).toFinset) := by
  sorry
```

Or were you suggesting to have a lemma that is still specialized to ConnectedComponent?

Reply:
This type-checks:
```lean
import Mathlib.Data.Set.Finite

lemma toFinset_iUnion {V W : Type*} [Fintype W] [Fintype V] [DecidableEq V] (f : W → Set V)
    [∀ w, Fintype (f w)] :
    Set.toFinset (⋃ (x : W), f x) =
    Finset.biUnion (Finset.univ : Finset W) (fun x => (f x).toFinset) := by
  sorry
```

Reply:
Thanks, managed to fill this in. I placed it close to the dependency on the needed Fintype instance, I couldn't really see a better place to put it.

Reply:
Looks good